### PR TITLE
Migrate FieldControllerBase to kotlin

### PR DESCRIPTION
**Pull request by lukstbit**
## Purpose / Description

Migrates the FieldControllerBase.java to kotlin following the migration guide.
Note for reviewers: After the migration using the tool I was left with the fields in FieldControllerBase as nullable and tens of changes to make(adding !!). Instead I did the kotlin cleanup for the nullable field to make them lateinit and reduced the number of changes(I reviewed the nullability in this class in a previous PR where I added the @ NotNull/@ Nullable annotations to the class). I plan to refactor this class even more in a future PR.

## How Has This Been Tested?

Ran the usual tests.

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by david-allison** [comment]
```
> Task :AnkiDroid:deleteSingleFileInGradle UP-TO-DATE
e: /home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/test/java/com/ichi2/anki/multimediacard/fields/BasicImageFieldControllerTest.kt: (93, 65): Cannot access 'mActivity': it is protected in 'BasicImageFieldController'
```

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/FieldControllerBase.kt [hunk 1]
```diff
@@ -12,61 +12,42 @@
  * This program is distributed in the hope that it will be useful, but WITHOUT ANY      *
  * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A      *
  * PARTICULAR PURPOSE. See the GNU General Public License for more details.             *
- *                                                                                      *
+ * *
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/FieldControllerBase.kt:15]: nit:revert

**Review by david-allison — CHANGES_REQUESTED** [review]
one nitpick, taking it on trust.

I read the description expecting the worst, but it's a tiny class and I fully support this.

My one request is to explain the change in the second PR's commit message

**Review by david-allison — APPROVED** [review]
Thanks so much!

**Comment by lukstbit** [comment]
@david-allison Sorry about the compilation error, that was a serious miss(I keep getting some unknown host exception for maven repo this morning).

**Comment by david-allison** [comment]
> @david-allison Sorry about the compilation error, that was a serious miss(I keep getting some unknown host exception for maven repo this morning).

Never an issue, just want to reiterate: thank you so much for all your time and effort, it's really appreciated!

**Review by mikehardy — APPROVED** [review]
nice!

**Pull request merged**